core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -248,6 +259,23 @@ public MetadataManager(
         verifyTypes();
 
         functionDecoder = new ResolvedFunctionDecoder(this::getType);
+
+        operatorCache = CacheBuilder.newBuilder()
+                .maximumSize(1000)
+                .build(CacheLoader.from(entry -> {
+                    String name = mangleOperatorName(entry.getKey());
+                    return resolveFunction(QualifiedName.of(name), fromTypes(entry.getValue()));
+                }));
+
+        coercionCache = CacheBuilder.newBuilder()
+                .maximumSize(1000)
```

Comment:
Also, question about sizing. I don't know what would be the proper size here.

eg how many different entries are getting inserted here during planning of the TPC-H and TPC-DS queries?

Reply:
We have similar cache in `io.trino.metadata.FunctionRegistry#FunctionRegistry`